tests/zkevm/test_worst_compute.py

```diff
@@ -741,6 +741,105 @@ def test_worst_unop(
     )
 
 
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize("key_mut", [True, False])
+@pytest.mark.parametrize("val_mut", [True, False])
+def test_worst_tload(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    key_mut: bool,
+    val_mut: bool,
+):
+    """Test running a block with as many TLOAD calls as possible."""
+    env = Environment()
+
+    start_key = 41
+    start_val = 43
+    code_prefix = Op.PUSH1(start_val) + Op.PUSH1(start_key) + Op.JUMPDEST
+
+    # If `key_mut` is True, we mutate the key on every iteration of the big loop by mutiplying by
+    # a constant that should produce a long chain of distinct values.
+    code_key_mut = Op.PUSH1(3) + Op.MUL if key_mut else Bytecode()
+
+    # If `val_mut` is True, we mutate the value that will be returned by the nex iteration of
+    # TLOADs.
+    code_val_mut = (
+        Op.DUP2  # MUL current value
+        + Op.PUSH1(3)  # MUL constant factor
+        + Op.MUL  # Calculate new value
+        + Op.SWAP2  # Save it
+        + Op.POP  # Cleanup
+        + Op.TSTORE(Op.DUP2, Op.DUP2)
+        if val_mut
+        else Bytecode()
+    )
+
+    # Note that both potential `code_key_mut` and `code_value_mut` are done at the end of
+    # big-loops to avoid creating too much influence in the benchmark results for measuring TLOADs.
+    code_suffix = code_key_mut + code_val_mut + Op.JUMP(len(code_prefix) - 1)
+
+    loop_iter = Op.POP(Op.TLOAD(Op.DUP1))
```

Comment:
Note, for more TLOAD (but will always push 0 to stack due to non-initialized storage) use `GAS` (2 gas) instead of `DUP1` (3 gas). This will also load different transient storage on each call.

Reply:
I think I would have a problem to make the value change work, right? Since I can't easily reproduce GAS on the TSTORE.

Reply:
For a stable value you could use `CALLVALUE` (2 gas) and store the to-be-pushed var in the callvalue?